### ISSUE TITLE
Hibernate reactive panache quickstart with UUIDGenerator throws error in native mode

### ISSUE THREAD

**Issue by Lahres-Dev** [quarkusio/quarkus#21217]
### Describe the bug

Running a customized hibernate-reactive-panache-quickstart project in native mode throws
`Could not interpret id generator strategy [org.hibernate.id.UUIDGenerator]` This looks similar to #1743

### Expected behavior

Should work like in dev or jvm mode

### Actual behavior

Does not run in native mode

### How to Reproduce?

1. Customize `hibernate-reactive-panache-quickstart` Fruit class from
<pre>
@Entity
@Cacheable
public class Fruit extends PanacheEntity {

    @Column(length = 40, unique = true)
    public String name;

    public Fruit() {
    }

    public Fruit(String name) {
        this.name = name;
    }

}
</pre>
to 
<pre>
@Entity
@Cacheable
public class Fruit extends PanacheEntityBase {

    @Id
    @GenericGenerator(name = "UUID", strategy = "org.hibernate.id.UUIDGenerator")
    @GeneratedValue(generator = "UUID")
    public UUID id;

    @Column(length = 40, unique = true)
    public String name;

    public Fruit() {
    }

    public Fruit(String name) {
        this.name = name;
    }
}
</pre>
2. Run project in native mode

### Output of `uname -a` or `ver`

MINGW64_NT-10.0-19043 XXXX 1 3.1.7-340.x86_64 2021-03-26 22:17 UTC x86_64 Msys

### Output of `java -version`

openjdk 11.0.11 2021-04-20 LTS OpenJDK Runtime Environment Corretto-11.0.11.9.1 (build 11.0.11+9-LTS) OpenJDK Server VM Corretto-11.0.11.9.1 (build 11.0.11+9-LTS, mixed mode, emulated-client)

### GraalVM version (if different from Java)

GraalVM 21.2.0 Java 11 CE (from native container build)

### Quarkus version or git rev

2.4.1.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.1 (05c21c65bdfed0f71a2f2ada8b84da59348c4c5d) Maven home: C:\Users\XXXX\.m2\wrapper\dists\apache-maven-3.8.1-bin\2l5mhf2pq2clrde7f7qp1rdt5m\apache-maven-3.8.1 Java version: 11.0.11, vendor: Amazon.com Inc., runtime: C:\copybase\jdk11.0.11_9 Default locale: en_US, platform encoding: Cp1252 OS name: "windows 10", version: "10.0", arch: "x86", family: "windows"

**Comment by geoand** [comment]
I was able to reproduce this.

What seems weird to me is that we have integration tests that use this same pattern and test native mode.

**Comment by geoand** [comment]
Done

**Comment by omar-zahid** [comment]
@DavideD  Any update on this issue? I'm getting similar error on native image.
![image](https://user-images.githubusercontent.com/57479337/167168680-64190cc5-7475-4485-8aa8-e5ef2b3da4cd.png)

**Comment by DavideD** [comment]
Sorry, I haven't had time to look into this yet

**Comment by summit-pgoncalves** [comment]
Getting the same exception here running in native mode, any new on this issue ? Using **2.10.1.Final** with **gradle 7.4.2**.

**Comment by philippegoncalves** [comment]
Getting the same exception here running in native mode, any new on this issue ? Using **2.10.1.Final** with **gradle 7.4.2**.

**Comment by philippegoncalves** [comment]
Same issue reproducible on latest version **2.11.2.** Something strange is, sometimes the native compile produce a binary that does not throw this error at startup, but most of the time it will. Note that we have a multi module project with a complex architecture, but this should not impact the native build compilation.

**Comment by philippegoncalves** [comment]
I just realized that the issue is because of using command **assemble** instead of **build** when using gradle as the build tool. Our bad, maybe it is worth it to specify on the documentation to only use the **build** command, as it is known to run the test as well, but not when building in native mode. I believe people coming from other gradle projects will be tempted to use the **assemble** command.
@HknLof are you using the **build** command or **assemble** ?

**Comment by Lahres-Dev** [comment]
When registering UUIDGenerator for reflection, everything works fine. I'm not sure if this is considered a workaround or intended:

```` java
@RegisterForReflection(targets = UUIDGenerator.class)
class MyReflectionConfiguration {}
````
See also: https://quarkus.io/guides/writing-native-applications-tips#registering-for-reflection

**Comment by philippegoncalves** [comment]
Yes it works by using the annotation.

**Comment by HknLof** [comment]
@philippegoncalves Have not tried in a while, but we are using `./gradlew build`. I will update Quarkus and gradlew versions and run the resulting image again. I will let you know about the outcome.


```
          ./gradlew build --info --no-daemon -x test \
                -Dquarkus.jib.platforms=linux/amd64,linux/arm64 \
                -Dquarkus.container-image.build=true \
                -Dquarkus.container-image.push=true \
                -Dquarkus.container-image.registry=ghcr.io \
                -Dquarkus.native.container-build=true \
                -Dquarkus.package.type=native \
                ...
```

**Comment by philippegoncalves** [comment]
On my side @HknLof I still need to add the @RegisterForReflection mentioned by @Lahres-Dev eventhough it is supposed to work wothout it based on https://github.com/quarkusio/quarkus/issues/1743

**Comment by Sanne** [comment]
Yes that's a good workaround. As mentioned we'll eventually rewrite the Hibernate Reactive extension to take such things into account, but we need to focus on Jakarta 10 + Hibernate ORM 6 first; so please use this workaround in the meantime.

**Comment by guoxunbo** [comment]
Hello. have the same exception in 2.13.3.final.

**Comment by geoand** [comment]
Is this still an issue with the latest versions of Quarkus?

**Comment by DavideD** [comment]
It seems to work fine now.

**Comment by geoand** [comment]
Thanks for checking
